### DISCUSSION TITLE
Syntactic idea for when variable is a tiddler title to access field values

### DISCUSSION THREAD

**Discussion by twMat** [TiddlyWiki/TiddlyWiki5#5656] [Ideas]
This does not work:

`[ ...sortby{<myvariable>!!myfield}]`

...because syntax does not permit a variable call inside the curly braces. Instead one has to split out the whole listwidget into a separate macro. But this is just not... elegant. **It is a syntactic detail dictating the whole code disposition**. It affects readability.

Considering that this is a common situation, couldn't this be solved more elegantly?

One idea would be to extend the variable call syntax: **It is common that the variable operand is a tiddler title**, so could it perhaps be extended to be a full text reference, i.e to refer to a fields content? `<<myvariable!!myfield>>`

Then the above filter would read `[ ...sortby<myvariable!!myfield>]`

Or maybe the new variable `<<__var__>>` syntax: `[ ...sortby<__myvariable!!myfield__>]`

**Side note**
I just realized that calling a variable has a lot of similarities to transclusions in the sense that you are, so to say, extracting what is inside it. So `<variablecall>` and `{textref}` seem to have potential to be merged syntactically... (TWX?)

**Comment by pmario** [comment]
> So <variablecall> and {textref} seem to have potential to be merged syntactically... (TWX?)

I don't think so. `<>` is the indication for the parser to create a tree, that can access a variable.
`{}` is the indicator to create a transclusion. 

Without this information, it wouldn't be possible to be 100% sure what to do. So additional info would be needed. Exactly the info we just removed.

**Comment by pmario** [comment]
Other operators allow us to specify a field. eg: `[ ...sortby:myfield[from-tiddler]]` or `[ ...sortby:myfield<from-tiddler-the-variable-points-to>]`  or `[ ...sortby:myfield{from-tiddler-the-tiddler-points-to}]` 

I think, that's consistent and covers all cases

**Reply by twMat** [reply]
>`[ ...sortby:myfield<from-tiddler-the-variable-points-to>]`

Ha! Indeed that does it. Thanks!

**Comment by pmario** [comment]
imo https://tiddlywiki.com/#enlist%20Operator%20(Examples) has the same problem. I'll add more operators here if I find some
